### PR TITLE
Update go version of workflow to 1.16

### PR DESCRIPTION
**Pull request by KanShiori**
### What problem does this PR solve?

Update go version of workflow to 1.16

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [x] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
None
```

### CODE REVIEW

File: .github/workflows/chaos.yaml [hunk 1]
```diff
@@ -11,10 +11,10 @@ jobs:
   pull:
     runs-on: ubuntu-latest
     steps:
-    - name: Set up Go 1.13
+    - name: Set up Go 1.16
       uses: actions/setup-go@v1
       with:
-        go-version: 1.13
+        go-version: 1.16
```
Comment (csuzhangxc) [inline .github/workflows/chaos.yaml:17]: we use 1.16.4 in our CI/CD, so I think we can update this to `1.16.4` too.

ref: https://github.com/actions/go-versions/blob/e891add10bdafdc3a5f66def262e646db7d66342/versions-manifest.json#L3

**Review by csuzhangxc — COMMENTED** [review]
rest LGTM.

BTW, [actions/setup-go@v2](https://github.com/actions/setup-go#v2) has been released, we may update to v2 later.

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Review by july2993 — APPROVED** [review]

**Pull request merged**